core/trino-spi/src/main/java/io/trino/spi/connector/RowChangeParadigm.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+public enum RowChangeParadigm
+{
+    ROW_CHANGE_NOT_SUPPORTED,
```

Comment:
remove

Reply:
I think I'd prefer to keep `ROW_CHANGE_NOT_SUPPORTED`, because some connectors might support changes for some kinds of tables but not others.  In this PR, Hive only supports MERGE for transactional tables.

Reply:
if ROW_CHANGE_NOT_SUPPORTED means not supported, how is it better than throwing?

> ome connectors might support changes for some kinds of tables but not others

that means we need to provide a table handle to the call where RowChangeParadigm is returned, but doesn't require having "null option" in the enum

Reply:
Removed and `HiveMetadata.getRowChangeParadigm()` changed to throw `NOT_SUPPORTED`.  Removed from `supporting-merge.rst`, and text changed to require throwing.